code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -444,15 +444,18 @@
 
 /mob/living/simple_animal/hostile/proc/DestroyObjectsInDirection(direction)
 	var/turf/T = get_step(targets_from, direction)
-	if(T && T.Adjacent(targets_from))
+	if(QDELETED(T))
+		return
+	if(T.Adjacent(targets_from))
 		if(CanSmashTurfs(T))
 			T.attack_animal(src)
-		for(var/obj/O in T)
-			if(O.density && environment_smash >= ENVIRONMENT_SMASH_STRUCTURES && !O.IsObscured())
+			return
+	for(var/obj/O in T.contents)
+		if(O.Adjacent(targets_from))
+			if((ismachinery(O) || isstructure(O)) && O.density && environment_smash >= ENVIRONMENT_SMASH_STRUCTURES && !O.IsObscured())
```

Comment:
Combine these ifs or at least use it to reduce the line length/complexity
```
if(O.Adjacent(targets_from) && (ismachinery(O) || isstructure(O)) ...
```
or
```
if(!O.Adjacent(targets_from))
  continue
if(...
```

Reply:
got it